lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -204,6 +182,59 @@ void DerefineOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// Aten__Is__Op
+//===----------------------------------------------------------------------===//
+
+OpFoldResult Aten__Is__Op::fold(ArrayRef<Attribute> operands) {
+  auto lhsType = self().getType();
+  auto rhsType = obj().getType();
+  // If either type is a NoneType, make it be the lhsType.
+  if (rhsType.isa<Basicpy::NoneType>())
+    std::swap(lhsType, rhsType);
+  // TODO: Implement and use subtype infra for this.
+  // If neither type is a subtype of the other, then the result is false.
+  if (lhsType.isa<Basicpy::NoneType>() && !rhsType.isa<Torch::OptionalType>())
+    return IntegerAttr::get(IntegerType::get(getContext(), 1), 0);
+  return nullptr;
+}
+
+//===----------------------------------------------------------------------===//
+// AtenLenTOp
+//===----------------------------------------------------------------------===//
+
+void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                             MLIRContext *context) {
+  patterns.add(+[](AtenLenTOp op, PatternRewriter &rewriter) {
```

Comment:
I don't believe the unary `+` is required for >C++11.

Reply:
Unfortunately it still seems to be needed on clang 9 on my machine. I think it has something to do with overload resolution / template matching and not the decay itself.

Reply:
clang 11 (latest on my machine) also has the same issue btw :(